clang warning points to a bug in code

The code (as you can see below) is wrong. The warning itself is a good explanation.

```
gpg2john.c:418:14: warning: comparison of constant -1 with expression of type
      'unsigned char' is always false
      [-Wtautological-constant-out-of-range-compare]
                if (buf[i] == EOF)
                    ~~~~~~ ^  ~~~
1 warning generated.
```

- I'm not sure if this is the code I commented recently. Anyway, since I failed to find an issue, I'm documenting it here.
- since Getc() is returning an integer, I see no reason to use an uint variable.
- I'll submit a patch if nobody disagrees/complains.

--- a/src/gpg2john.c
+++ b/src/gpg2john.c
@@ -407,16 +407,17 @@ dump(unsigned len)
 public int
 give(const unsigned len, unsigned char *buf, const unsigned buf_size)
 {
-	int i;
+	int i, tmp;
 
 	if (len > buf_size)
 		warn_exit("Bad parameter: give(len=%u, buf=%p, buf_size=%u), len can not be bigger than buf_size.",
 			len, buf, buf_size);
 
 	for (i = 0; i < len; i++) {
-		buf[i] = Getc();
-		if (buf[i] == EOF)
+		tmp = Getc();
+		if (tmp == EOF)
 			return -1;
+		buf[i] = tmp;
 	}
 	return len;
 }

